go.mod

```diff
@@ -71,6 +71,7 @@ require (
 	github.com/prometheus/client_golang v1.11.0
 	github.com/prometheus/client_model v0.2.0
 	github.com/prometheus/common v0.28.0
+	github.com/quasilyte/go-ruleguard/dsl v0.3.10
```

Comment:
do we *have* to reference this from the root and library `go.mod` files? there's no way we can reference it from `hack/tools/go.mod` instead?

Reply:
last time I've tried it requires the code to be linted to import this dependency ... that makes me think we also have to import it in all staging repos :thinking:  we should verify it

Reply:
ic, last commit adds it as dependency of the staging repos

Reply:
nope - ruleguard itself uses Go tooling, and if the module doesn't have the DSL, it can't parse the rules file.  I filed bugs and slugged this out with the ruleguard and golangci folks.

Let me ask a different question - is it important that we check-in the go.mod for each staging repo? If we didn't check it in, just generated it at publishing time, we would dodge a LOT of problems...

Reply:
is it worth  changing everything just to use ruleguard ?